store.go

```diff
@@ -2103,117 +2090,98 @@ func (s *store) ContainerSize(id string) (int64, error) {
 			if err != nil {
-				return fmt.Errorf("reading size of big data item %q for container %q: %w", name, id, err)
+				return -1, fmt.Errorf("reading size of big data item %q for container %q: %w", name, id, err)
 			}
 			size += n
 		}
 
 		// Count the size of our container directory and container run directory.
 		n, err := directory.Size(cdir)
 		if err != nil {
-			return err
+			return -1, err
 		}
 		size += n
 		n, err = directory.Size(rdir)
 		if err != nil {
-			return err
+			return -1, err
 		}
 		size += n
 
-		res = size
-		return nil
+		return size, nil
 	})
-	return res, err
 }
 
 func (s *store) ListContainerBigData(id string) ([]string, error) {
-	if err := s.containerStore.startReading(); err != nil {
-		return nil, err
-	}
-	defer s.containerStore.stopReading()
-
-	return s.containerStore.BigDataNames(id)
+	return readContainerStore(s, func() ([]string, error) {
+		return s.containerStore.BigDataNames(id)
+	})
 }
 
 func (s *store) ContainerBigDataSize(id, key string) (int64, error) {
-	var res int64 = -1
-	err := s.writeToContainerStore(func() error { // Yes, BigDataSize requires a write lock.
-		var err error
-		res, err = s.containerStore.BigDataSize(id, key)
-		return err
+	return writeToContainerStore(s, func() (int64, error) { // Yes, BigDataSize requires a write lock.
+		return s.containerStore.BigDataSize(id, key)
 	})
-	return res, err
 }
 
 func (s *store) ContainerBigDataDigest(id, key string) (digest.Digest, error) {
-	var res digest.Digest
-	err := s.writeToContainerStore(func() error { // Yes, BigDataDigest requires a write lock.
-		var err error
-		res, err = s.containerStore.BigDataDigest(id, key)
-		return err
+	return writeToContainerStore(s, func() (digest.Digest, error) { // Yes, BigDataDigest requires a write lock.
+		return s.containerStore.BigDataDigest(id, key)
 	})
-	return res, err
 }
 
 func (s *store) ContainerBigData(id, key string) ([]byte, error) {
-	if err := s.containerStore.startReading(); err != nil {
-		return nil, err
-	}
-	defer s.containerStore.stopReading()
-	return s.containerStore.BigData(id, key)
+	return readContainerStore(s, func() ([]byte, error) {
+		return s.containerStore.BigData(id, key)
+	})
 }
 
 func (s *store) SetContainerBigData(id, key string, data []byte) error {
-	return s.writeToContainerStore(func() error {
-		return s.containerStore.SetBigData(id, key, data)
+	_, err := writeToContainerStore(s, func() (void, error) {
+		return void{}, s.containerStore.SetBigData(id, key, data)
 	})
+	return err
 }
 
 func (s *store) Exists(id string) bool {
-	var res = false
-
-	if done, _ := s.readAllLayerStores(func(store roLayerStore) (bool, error) {
+	if res, done, _ := readAllLayerStores(s, func(store roLayerStore) (bool, bool, error) {
 		if store.Exists(id) {
-			res = true
-			return true, nil
+			return true, true, nil
 		}
-		return false, nil
+		return false, false, nil
 	}); done {
-		return res
+		return res // false if err != nil
```

Comment:
Which `err` is this referring to, here and in various similar comments where the closure is calling `Exists()`?

Reply:
The one that was assigned to `_` when calling the locking wrapper…

It would be possible to explain that that in those comments, but at that point it seems simpler to just make the code explicit `if err { return false }; if found { return true }` now. So updated that way.